### PR TITLE
simplify control flow in prefix comparison

### PR DESCRIPTION
**Pull request by arriqaaq**

### CODE REVIEW

File: src/art.rs [hunk 1]
```diff
@@ -746,53 +746,47 @@ impl<P: KeyTrait, V: Clone> Node<P, V> {
...
+        let shared_prefix_length = cur_node_prefix.longest_common_prefix(remaining_key_suffix);
 
         // Create a new key that represents the remaining part of the current node's prefix after the common prefix.
-        let mut new_key = cur_node_prefix.prefix_after(longest_common_prefix);
-        if new_key.is_empty() {
+        let mut remaining_node_suffix = cur_node_prefix.prefix_after(shared_prefix_length);
+        if remaining_node_suffix.is_empty() {
             // In the case where the current node's prefix is shorter than the key,
             // the suffix comes from the key_prefix. For example, cur_node.prefix="key"
             // and key="keyboard", then new_key should be "board".
```
Comment (gsserge) [inline src/art.rs:770]: `new_key` should be changed to `remaining_node_suffix `.

File: src/art.rs [hunk 2]
```diff
@@ -802,119 +796,128 @@ impl<P: KeyTrait, V: Clone> Node<P, V> {
...
-            let new_node = cur_node.replace_child(k, new_child);
-            return Arc::new(new_node);
-        }
+        // Case 2: Handle prefix match scenarios
+        let cur_prefix_len = cur_node.prefix().len();
+        let key_prefix_len = key_prefix.len();
+
+        match cur_prefix_len.cmp(&key_prefix_len) {
+            // Case 2a: Exact prefix match
+            std::cmp::Ordering::Equal => {
```
Comment (gsserge) [inline src/art.rs:828]: Let's do `use std::cmp::Ordering` and make it easy on the eyes by using `Ordering::Equal` and the others.

Comment (arriqaaq) [reply src/art.rs]: Done!

File: src/art.rs (hunk 1 above)
Comment (arriqaaq) [reply src/art.rs]: Done!

**Pull request merged**